testindex/summary.py

```diff
@@ -21,39 +21,46 @@ def add_warning(self, msg):
 class Summary(object):
     """This class summarizes the tests"""
 
-    _summary = {}
-
-    def __init__(self):
+    def __init__(self, summary_dump):
+        self.global_errors = []
+        self._summary = {}
+        self._summary_dump = summary_dump
         pass
 
     @staticmethod
     def _get_entry_hash(file_name, entry):
 
         return hashlib.sha256(file_name + str(entry)).hexdigest()
 
-    @staticmethod
-    def pakage_private_get_summary_object(file_name, entry):
+    def get_summary_object(self, file_name, entry):
 
-        if file_name not in Summary._summary:
-            Summary._summary[file_name] = {}
+        if file_name not in self._summary:
+            self._summary[file_name] = {}
 
         entry_hash = Summary._get_entry_hash(file_name, entry)
 
-        if entry_hash not in Summary._summary[file_name]:
-            Summary._summary[file_name][entry_hash] = {
+        if entry_hash not in self._summary[file_name]:
+            self._summary[file_name][entry_hash] = {
                 "entry": str(entry),
                 "errors": [],
                 "warnings": []
             }
 
-        return Summary._summary[file_name][entry_hash]
+        return self._summary[file_name][entry_hash]
 
-    @staticmethod
-    def print_summary():
+    def print_summary(self):
 
         print "\n####################### SUMMARY ##################\n"
 
-        for file_name, entries in Summary._summary.iteritems():
+        print "\nGLOBAL ERRORS:\n"
+        if len(self.global_errors) == 0:
+            print "NONE\n"
+        else:
+            for err in self.global_errors:
+                print "**E " + err
+            print
```

Comment:
Maybe just do:
```python
print "**E " + err + "\n"
```
instead of a blank print that would also just insert a newline? 

Not a big deal if we don't. :smile: